sdk/python/symbolchain/nem/MessageEncoder.py

```diff
@@ -0,0 +1,96 @@
+import secrets
+
+import cryptography
+
+from symbolchain.Cipher import AesCbcCipher, AesGcmCipher
+from symbolchain.CryptoTypes import PublicKey
+from symbolchain.nc import Message, MessageType
+from symbolchain.nem.KeyPair import KeyPair
+from symbolchain.nem.SharedKey import SharedKey
+
+
+class MessageEncoder:
+	"""Encrypts and encodes messages between two parties."""
+
+	def __init__(self, key_pair: KeyPair):
+		"""Creates message encoder around key pair."""
+		self.key_pair = key_pair
+
+	def _decode_aes_gcm(self, recipient_public_key, encoded_message):
+		# pylint: disable=duplicate-code
+		GCM_IV_SIZE = 12  # pylint: disable=invalid-name
```

Comment:
why duplicate?

Reply:
_decode_aes_gcm "looks"  same in both, but is using different **types**, I was thinking about extracting it earlier, but not sure _where_ (i.e. putting it somewhere inside `symbolchain.foo` is probably not the best idea, as such helper should not be used directly)

Reply:
i would probably refactor. can create new submodule like `internal` or `impl` or `private